core/crypto/client_impl.go

```diff
@@ -44,31 +46,39 @@ func (client *clientImpl) NewChaincodeDeployTransaction(chaincodeDeploymentSpec
 	}
 
 	// Get next available (not yet used) transaction certificate
-	tCert, err := client.tCertPool.GetNextTCert()
+	tCerts, err := client.tCertPool.GetNextTCerts(1)
 	if err != nil {
 		client.error("Failed getting next transaction certificate [%s].", err.Error())
```

Comment:
While usually such errors are printed with the/a stack trace - I totally agree with you. Tried to make these more intuitive and contextual (while unique). Thanks for this. Please see: [3f43948](https://github.com/hyperledger/fabric/pull/1714/commits/3f43948b10e50a560ecd852b7ca09e46e8e1ceb2)